src/cml.js

```diff
@@ -159,9 +182,63 @@ class CML {
       ? ''
       : '![CML watermark](https://raw.githubusercontent.com/iterative/cml/master/assets/watermark.svg)';
 
-    const report = `${userReport}\n\n${watermark}`;
+    let userReport = testReport;
+    try {
+      if (!userReport)
+        userReport = await fs.promises.readFile(markdownFile, 'utf-8');
+    } catch (err) {
+      if (!watch) throw err;
+    }
+
+    let report = `${userReport}\n\n${watermark}`;
     const drv = getDriver(this);
 
+    const publishLocalFiles = () => async (tree) => {
```

Comment:
Not a fan of this function returns a function syntax /off-topic

Reply:
Same. Do you know any better alternative?

Reply:
not one that I think others would like.